src/common/src/array/jsonb_array.rs

```diff
@@ -69,29 +70,44 @@ impl ArrayBuilder for JsonbArrayBuilder {
         for bit in other.bitmap.iter() {
             self.bitmap.append(bit);
         }
-        self.data.extend_from_slice(&other.data);
+        for value in other.data.as_array().unwrap().iter() {
+            self.builder.add_value(value);
+        }
     }
 
     fn pop(&mut self) -> Option<()> {
-        self.data.pop().map(|_| self.bitmap.pop().unwrap())
+        self.bitmap.pop()?;
+        self.builder.pop();
+        Some(())
     }
 
     fn len(&self) -> usize {
         self.bitmap.len()
     }
 
-    fn finish(self) -> Self::ArrayType {
+    fn finish(mut self) -> Self::ArrayType {
+        self.builder.end_array();
         Self::ArrayType {
             bitmap: self.bitmap.finish(),
-            data: self.data,
+            data: self.builder.finish(),
         }
     }
 }
 
-impl JsonbArrayBuilder {
-    pub fn append_move(&mut self, value: JsonbVal) {
-        self.bitmap.append(true);
-        self.data.push(*value.0);
+impl JsonbArray {
+    /// Loads a `JsonbArray` from a protobuf array.
+    ///
+    /// See also `JsonbArray::to_protobuf`.
+    pub fn from_protobuf(array: &PbArray) -> ArrayResult<ArrayImpl> {
+        ensure!(
+            array.values.len() == 1,
+            "Must have exactly 1 buffer in a jsonb array"
+        );
+        let arr = JsonbArray {
+            bitmap: array.get_null_bitmap()?.into(),
+            data: jsonbb::Value::from_bytes(&array.values[0].body),
```

Comment:
I don't think it's necessary because the bytes must be serialized from another json array. If the data is corrupted (which is most likely a bug), it will panic when accessing the array.

Reply:
Then why do we check `array.values.len() == 1` 😂

Reply:
Just copied from other array. 😂